### PR TITLE
API spec changes for PSTN Dialout for Rooms

### LINKED ISSUE #25486: [IC3 ACS Rooms ] API Review
New API Review meeting has been requested.

**Service Name**: IC3 ACS Rooms 
**Review Created By**: Shikha Avasthi
**Review Date**: 09/07/2023 04:00 PM PT
**Onboarding Record**: 
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/25451/files
**Hero Scenarios Link**: [here](https://microsoft.sharepoint-df.com/:w:/t/IC3SDK/Edo7HPoKlu9DnwmDdWpZEDkBN1slsTpfx28HEmsbvY_3yQ?e=H2gSH2)
**Core Concepts Doc Link**: [here](https://microsoft.sharepoint-df.com/:w:/t/IC3SDK/Edo7HPoKlu9DnwmDdWpZEDkBN1slsTpfx28HEmsbvY_3yQ?e=H2gSH2)

**Description**: Currently Rooms only supports VOIP calls. These change is for enabling dial out to a PSTN number in a Rooms call.

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=8aa2123f-0442-ee11-bdf3-000d3a3412d3)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by shwali-msft**
API spec changes for PSTN Dialout for Rooms

fix https://github.com/Azure/azure-rest-api-specs/issues/25486

### CODE REVIEW

**Comment by JeffreyRichter** [comment]
I see new required properties here: https://github.com/Azure/azure-rest-api-specs/pull/25451/checks?check_run_id=16123854044
Do these have to be required? It is a breaking change which will have to be justified.

**Comment by JeffreyRichter** [comment]
Adding the new flag property is fine but the description is not great. How about: Set this flag to true if, at the time of the call, dial out to a PSTN number in enabled in a particular room.

**Comment by shwali-msft** [comment]
> Adding the new flag property is fine but the description is not great. How about: Set this flag to true if, at the time of the call, dial out to a PSTN number in enabled in a particular room.

Updated description as suggested.

**Comment by shwali-msft** [comment]
> I see new required properties here: https://github.com/Azure/azure-rest-api-specs/pull/25451/checks?check_run_id=16123854044 Do these have to be required? It is a breaking change which will have to be justified.

Yes, this needs to be added as a required property in RoomModel object. I have scheduled an API review with ARB for getting this reviewed.

File: specification/communication/data-plane/Rooms/preview/2023-10-30-preview/communicationservicesrooms.json [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+          "example": "2021-09-07T07:55:41Z"
+        },
+        "validUntil": {
+          "format": "date-time",
+          "description": "The timestamp from when the room can no longer be joined. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`. The default value is the current date time plus 180 days.",
+          "type": "string",
+          "example": "2021-09-08T15:55:41Z"
+        },
+        "enablePstnDialout": {
+          "description": "Set this flag to true if, at the time of the call, dial out to a PSTN number in enabled in a particular room.",
```
Comment (DominikMe) [inline specification/communication/data-plane/Rooms/preview/2023-10-30-preview/communicationservicesrooms.json:410]:
```suggestion
          "description": "Set this flag to true if, at the time of the call, dial out to a PSTN number is enabled in a particular room.",
```

**Review by JeffreyRichter — APPROVED** [review]

**Comment by anuchandy** [comment]
Approving the model validation, as the same were identified as false positive and confirmed with Azure Board in an earlier pr https://github.com/Azure/azure-rest-api-specs/pull/22540

**Pull request merged**